trufflesuite/truffle
# Add Hardhat plugin for Truffle Dashboard

## PR description

This PR adds the Hardhat plugin for Truffle Dashboard. It extends `npx hardhat compile` by sending the compiled artifacts (which utilizes `@truffle/from-hardhat`'s `prepareCompilations`) to the Dashboard via the message bus. Subsequent transactions are now able to display the decoded request, etc given users that extra of degree of confidence before they sign transactions.

## Testing instructions

- Create a new Hardhat project (or use an existing one if available)…[template](https://github.com/amanusk/hardhat-template) if needed
- Follow the steps in the [README](https://github.com/trufflesuite/truffle/tree/feat/dashboard-hardhat-plugin/packages/dashboard-hardhat-plugin) to setup + configure the plugin for use in your project
- You'll also need to both compile (for which you should see additional console output stating the compilation has been sent to the dashboard) and send a transaction, upon which you should now see [decoded request](https://raw.githubusercontent.com/trufflesuite/truffle/feat/dashboard-hardhat-plugin/packages/dashboard-hardhat-plugin/assets/truffle-dashboard-screenshot.jpg).


## Documentation


## Breaking changes and new features
